mev-boost-rs/src/relay.rs

```diff
@@ -0,0 +1,78 @@
+use std::ops::Deref;
+
+use beacon_api_client::Client as BeaconClient;
+use ethereum_consensus::{
+    crypto::Error as CryptoError, primitives::BlsPublicKey, serde::try_bytes_from_hex_str,
+};
+use url::Url;
+
+use mev_rs::blinded_block_provider::Client;
+
+#[derive(Clone, Debug)]
+pub struct RelayEndpoint {
+    url: Url,
+    public_key: BlsPublicKey,
+}
+
+impl TryFrom<Url> for RelayEndpoint {
+    type Error = CryptoError;
+
+    fn try_from(url: Url) -> Result<Self, Self::Error> {
+        let public_key = try_bytes_from_hex_str(url.username())?;
+        let public_key = BlsPublicKey::try_from(&public_key[..])?;
+
+        Ok(Self { url, public_key })
+    }
+}
+
+#[derive(Clone)]
+pub struct Relay {
+    api: Client,
+    pub(crate) public_key: BlsPublicKey,
+}
+
+impl Deref for Relay {
+    type Target = Client;
+
+    fn deref(&self) -> &Self::Target {
+        &self.api
+    }
+}
+
+impl From<RelayEndpoint> for Relay {
+    fn from(value: RelayEndpoint) -> Self {
+        let RelayEndpoint { url, public_key } = value;
+        Self { api: Client::new(BeaconClient::new(url)), public_key }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use ethereum_consensus::crypto::SecretKey;
+
+    const URL: &str = "https://relay.com";
+
+    #[test]
+    fn parse_relay_endpoint() {
+        let mut rng = rand::thread_rng();
+        let sk = SecretKey::random(&mut rng).unwrap();
+        let public_key = sk.public_key();
+        let public_key_hex = format!("{:#x}", *public_key);
```

Comment:
```suggestion
        let public_key_hex = format!("{public_key}");
```

Reply:
ah we can just go with `to_string` here. I think that's simpler